indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java

```diff
@@ -410,4 +422,90 @@ public static Optional<Contacts> mapContactInfo(List<CIContactPropertyType2> con
         }
     }
 
+    public static List<ContactsModel> mapContactsFromOrg(CIResponsibilityType2 ciResponsibility, CIOrganisationType2 organisation) {
+
+        Optional<MapperUtils.Contacts> org = MapperUtils.mapContactInfo(organisation.getContactInfo());
+        return new ArrayList<>(organisation
+                .getIndividual()
+                .stream()
+                .map(individual -> {
+                    ContactsModel invContactsModel = ContactsModel.builder().build();
+                    invContactsModel.setName(MapperUtils.mapContactsName(individual));
+                    invContactsModel.setPosition(MapperUtils.mapContactsPosition(individual));
+                    invContactsModel.setRoles(MapperUtils.mapContactsRole(ciResponsibility));
+                    invContactsModel.setOrganization(organisation.getName().getCharacterString().getValue().toString());
+
+                    Optional<MapperUtils.Contacts> inv = MapperUtils.mapContactInfo(individual.getCIIndividual().getContactInfo());
+                    MapperUtils.Contacts i = org.orElse(null);
+
+                    // Address
+                    if (inv.isPresent() && !inv.get().getAddresses().isEmpty()) {
+                        invContactsModel.setAddresses(inv.get().getAddresses());
+                    } else {
+                        invContactsModel.setAddresses(i != null ? i.getAddresses() : null);
+                    }
+                    // Email
+                    if (inv.isPresent() && !inv.get().getEmails().isEmpty()) {
+                        invContactsModel.setEmails(inv.get().getEmails());
+                    } else {
+                        invContactsModel.setEmails(i != null ? i.getEmails() : null);
+                    }
+                    // Phone
+                    if (inv.isPresent() && !inv.get().getPhones().isEmpty()) {
+                        invContactsModel.setPhones(inv.get().getPhones());
+                    } else {
+                        invContactsModel.setPhones(i != null ? i.getPhones() : null);
+                    }
+                    // Online Resources
+                    // Phone
+                    if (inv.isPresent() && !inv.get().getOnlineResources().isEmpty()) {
+                        invContactsModel.setLinks(inv.get().getOnlineResources());
+                    } else {
+                        invContactsModel.setLinks(i != null ? i.getOnlineResources() : null);
+                    }
```

Comment:
unnecessary duplicated `.isPresent()`

Reply:
I understand you have `.isEmpty()` for each of the field, but you will also need to have null check first for each of them, null isn't the same with empty array.

Reply:
try not to use abbreviation for variable names where possible. `i` should be for stuff like the `for` loop e.g `for int i = 0; ...` , very hard to recall a few years after , `i` then `inv` ?

Reply:
instance, index, input, individual, etc. ? they can all be interpreted by `i`